security/leaks.js

```diff
@@ -0,0 +1,140 @@
+window.addEventListener('load', async () => {
+    const sections = {};
+    ['added', 'removed', 'changed'].forEach((s) => {
+        sections[s] = document.getElementById(s);
+    });
+
+    function addBullet (html, section) {
+        const li = document.createElement('li');
+        li.innerHTML = html;
+        sections[section].appendChild(li);
+    }
+
+    function collectProps () {
+        const objCache = new Set();
+        // skip these
+        objCache.add(document);
+        objCache.add(window.self);
+        objCache.add(document.activeElement);
+        objCache.add(event?.currentTarget);
+        const recursionAllowed = new Set();
+        recursionAllowed.add('navigator');
+
+        function extractObjectProps (obj, bc = []) {
+            const props = {};
+            const proplist = [];
+            for (const prop in obj) {
+                proplist.push(prop);
+            }
+            proplist.sort().forEach((prop) => {
+                if (bc.length === 0 && prop === 'collectProps') {
+                    return;
+                }
+                try {
+                    const desc = Object.getOwnPropertyDescriptor(obj, prop) || {};
+                    desc.type = typeof obj[prop];
+                    if (desc.type === 'function') {
+                        desc.value = obj[prop].toString();
+                    } else if (desc.type === 'object') {
+                        if (!objCache.has(obj[prop]) || recursionAllowed.has(prop)) {
+                            objCache.add(obj[prop]);
+                            desc.properties = extractObjectProps(obj[prop], [...bc, prop]);
```

Comment:
should we keep track of `depth` for recursion? I guess we have that objCache, so maybe we are not too worried about going too deep?